pyanaconda/modules/payload/dnf/packages/packages.py

```diff
@@ -0,0 +1,377 @@
+        self._timeout = None
+        self.timeout_changed = Signal()
+        self._retries = None
+        self.retries_changed = Signal()
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(DNF_PACKAGES.object_path, PackagesHandlerInterface(self))
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        log.debug("Processing kickstart data...")
+        packages = data.packages
+
+        self.set_no_core_group(packages.nocore)
+        self.set_default_package_set(packages.default)
+
+        self.set_environment(packages.environment)
+        self.set_group_list(packages.groupList)
+        self.set_package_list(packages.packageList)
+
+        self.set_exclude_package_list(packages.excludedList)
+        self.set_exclude_group_list(packages.excludedGroupList)
+
+        self.set_exclude_docs(packages.excludeDocs)
+        self.set_exclude_weakdeps(packages.excludeWeakdeps)
+
+        if packages.handleMissing == KS_MISSING_IGNORE:
+            self.set_ignore_missing(True)
+        else:
+            self.set_ignore_missing(False)
+
+        self.set_install_languages(packages.instLangs)
+        self.set_multi_lib(packages.multiLib)
+        self.set_timeout(packages.timeout)
+        self.set_retries(packages.retries)
+
+    def setup_kickstart(self, data):
+        """Setup the kickstart data."""
+        log.debug("Generating kickstart data...")
+        packages = data.packages
+
+        packages.nocore = self.no_core_group
+        packages.default = self.default_package_set
+
+        packages.environment = self.environment
+        packages.groupList = self.group_list
+        packages.packageList = self.package_list
+
+        packages.excludedList = self.exclude_package_list
+        packages.excludedGroupList = self.exclude_group_list
+
+        packages.excludeDocs = self.exclude_docs
+        packages.excludeWeakdeps = self.exclude_weakdeps
+        packages.handleMissing = KS_MISSING_IGNORE if self.ignore_missing else KS_MISSING_PROMPT
+        packages.instLangs = self.install_languages
+        packages.multiLib = self.multi_lib
+        packages.timeout = self.timeout
+        packages.retries = self.retries
+
+        # The empty packages section won't be printed without seen set to True
+        packages.seen = True
+
+    @property
+    def no_core_group(self):
+        """Should the core group be installed?
+
+        :rtype: Bool
+        """
+        return self._no_core_group
+
+    def set_no_core_group(self, no_core_group):
+        """Set if the core group should be installed.
+
+        :param no_core_group: True if the core group should be installed
+        :type no_core_group: Bool
+        """
+        self._no_core_group = no_core_group
+        self.no_core_group_changed.emit(self._no_core_group)
+        log.debug("The no_core_group has changed")
+
+    @property
+    def default_package_set(self):
```

Comment:
I would guess that the property returns a set of default packages based on the name.

Reply:
Fixed.